pkg/upgrade/hub.go

```diff
@@ -120,3 +130,48 @@ func (b *FListInfo) Version() (semver.Version, error) {
 
 	return b.extractVersion(b.Name)
 }
+
+// Absolute returns the actual flist name
+func (b *FListInfo) Absolute() string {
+	name := b.Name
+	if b.Type == "symlink" {
+		name = b.Target
+	}
+
+	return filepath.Join(b.Repository, name)
+}
+
+// Files gets the list of the files of an flist
+func (b *FListInfo) Files() ([]File, error) {
```

Comment:
And this method also doesn't need to be exposed

Reply:
While idd it's used by the Updater only, i still find it useful to have it exposed, just for the sake of the API completion.

Same reason I also exposed the `File` struct. A user of the pkg, will find a usable Hub client

Reply:
Well I tend to try to keep the public API as small as possible.
If we want to provide a proper Hub lib, then we should extract it into its own package separate from the Upgrade type so it can really be reused by other.

Reply:
Okay, this makes sense. but in that case I will un-export the entire client then.